test/ng/httpSpec.js

```diff
@@ -781,6 +781,26 @@ describe('$http', function() {
         $httpBackend.flush();
       });
 
+      it('should expose a copy of a config object to header functions', function() {
+        var config = {
+          foo: 'Rewritten',
+          headers: {'Accept': function(config) {
+            config.bar = 'baz';
+            return config.foo;
+          }}
+        };
+
+        function checkHeaders(headers) {
+          return headers['Accept'] == 'Rewritten';
+        }
+
+        $httpBackend.expect('GET', '/url', undefined, checkHeaders).respond('');
+        $http.get('/url', config);
+        $httpBackend.flush();
+
+        expect(config.bar).toBeUndefined();
+      });
+
```

Comment:
let me check ngMock one sec... But in spite of that, I think you should have a different spec for that, because it's not explicitly testing that, it's explicitly testing that the passed in config object is a shallow copy